### PR TITLE
Make package pass R CMD Check

### PR DESCRIPTION
**Pull request by rcannood**
Hello @gonzaparra,

@zouter and I made a few modifications to the merlot package in order for it to be able to pass a R CMD check without any errors or warnings. 

I hope these changes will benefit you and other merlot users.

Kind regards,
Robrecht

Changelist:
* .Rbuildignore: Added an .Rbuildignore so that unnecessary files don't get built into the R package
* .gitignore: Added .Rhistory to the .gitignore
* .travis.yml: merlot can now be automatically built on travis-ci.org, if you like. Example: [https://travis-ci.org/dynverse/merlot](https://travis-ci.org/dynverse/merlot)
* DESCRIPTION: Packages were moved from Depends to Imports. If you add packages to Depends and run `library(merlot)`, many other libraries will be loaded along with it. Instead, `#' @importFrom packageName functionName` was used every time a function from a different package is used.
* DESCRIPTION: Remotes mentions what other github repository ElPiGraph.R comes from
* R/*.R: Fixed a lot of documentation
* R/zzz.R and inst/make: Upon installation, a virtualenv is created in the R package, during which any depending python packages are installed
* R/hello.R: Removed example file

### CODE REVIEW

**Comment by rcannood** [comment]
I noticed a problem with the R/zzz.R and the inst/make file -- during an R CMD CHECK these could not be found properly. Now, during installation, a virtualenv is created in `~/R/x86_64-redhat-linux-gnu-library/3.5/merlot/venv`, so that the R package is less dependant on the local Python installation.

**Comment by rcannood** [comment]
Hello @gonzaparra,

Have you had the time to look at the pull request?

Kind regards,
Robrecht

**Comment by rcannood** [comment]
Hello Niko,

The virtualenv was created when the package was installed. This would install the necessary python packages within the merlot package, so that it does not depend on the hosts installation of python in order to ensure everything is in working order.

However, we don't need this anymore, since dynmethods switched to all docker containers; this way we can ensure that merlot has the right packages installed inside of the docker at all times, so I removed venv from our pull request.

Kind regards,
Robrecht

File: R/ElasticPrincipalTree.R [hunk 1]
```diff
@@ -543,97 +560,6 @@ CalculateElasticTreeConstrained <- function(ScaffoldTree, N_yk=150, start_N_yk=1
   return (ElasticTree)
 }
 
-#' Duplicate Elastic Tree Nodes
```
Comment (galicae) [inline R/ElasticPrincipalTree.R]: @rcannood my only remaining objection is the (complete) removal of this function. It is a cost-effective way of adding more nodes to the tree (adding more detail in gene expression smoothing) without fitting a whole new tree. Other than that I think the rest of the pull request is great!

Comment (galicae) [reply R/ElasticPrincipalTree.R]: Should I just make the commit myself at the dynverse repo? How is the "correct" way to proceed here? :D

Comment (rcannood) [reply R/ElasticPrincipalTree.R]: I didn't actually remove the function, it's just that it's in the source code twice, namely:
* [ElasticPrincipalTree.R#L191](https://github.com/soedinglab/merlot/blob/master/R/ElasticPrincipalTree.R#L191)
* [ElasticPrincipalTree.R#L553](https://github.com/soedinglab/merlot/blob/master/R/ElasticPrincipalTree.R#L553)

Comment (rcannood) [reply R/ElasticPrincipalTree.R]: I didn't actually remove the function, it's just that it's in the source code twice, namely:
* [ElasticPrincipalTree.R#L191](https://github.com/soedinglab/merlot/blob/master/R/ElasticPrincipalTree.R#L191)
* [ElasticPrincipalTree.R#L553](https://github.com/soedinglab/merlot/blob/master/R/ElasticPrincipalTree.R#L553)

Is it okay to remove the duplicated function?

Comment (rcannood) [reply R/ElasticPrincipalTree.R]: I didn't actually remove the function, it's just that it's in the source code twice, namely:
* [ElasticPrincipalTree.R#L191](https://github.com/soedinglab/merlot/blob/master/R/ElasticPrincipalTree.R#L191)
* [ElasticPrincipalTree.R#L553](https://github.com/soedinglab/merlot/blob/master/R/ElasticPrincipalTree.R#L553)

Is it okay to remove the duplicated function?

**Comment by rcannood** [comment]
I didn't actually remove the function, it's just that it's in the source code twice, namely:
* [ElasticPrincipalTree.R#L191](https://github.com/soedinglab/merlot/blob/master/R/ElasticPrincipalTree.R#L191)
* [ElasticPrincipalTree.R#L553](https://github.com/soedinglab/merlot/blob/master/R/ElasticPrincipalTree.R#L553)

Is it okay to remove the duplicated function?

File: R/ElasticPrincipalTree.R (hunk 1 above)
Comment (galicae) [reply R/ElasticPrincipalTree.R]: Oh wow. I have no idea how we missed this, it is probably the relic of some experimentation :smile: I closed the change request, I will go over everything one last time and accept, probably later today. Thanks again for the great work!

**Pull request merged**